client/service/src/lib.rs

```diff
@@ -314,6 +315,7 @@ mod waiting {
 fn start_rpc_servers<R>(
 	config: &Configuration,
```

Comment:
Is there a reason why we can't put the ID provider in the `Configuration` struct instead of adding a new param?

Reply:
cool, I thought it was generated for clap/structopt but cleary not good idea.